src/gen_smtp_server_session.erl

```diff
@@ -756,62 +764,73 @@ handle_error(Kind, Details, #state{module=Module, callbackstate = OldCallbackSta
 			State
 	end.
 
--spec parse_encoded_address(Address :: binary()) -> {binary(), binary()} | 'error'.
-parse_encoded_address(<<>>) ->
```

Comment:
most of the diff in `parse_encoded_address` is just me replacing the tuple with the record (except one clause I added and changing `list_to_binary` to `unicode:characters_to_binary`)